src/common/expression/Expression.h

```diff
@@ -223,6 +225,17 @@ class Expression {
 
 std::ostream& operator<<(std::ostream& os, Expression::Kind kind);
 
+struct PropertyTracker {
+  std::unordered_map<std::string, std::unordered_map<TagID, std::unordered_set<std::string>>>
```

Comment:
Why use TagID rather than TagName?

Reply:
Because the plan node such as `AppendVertices` or `Traverse` 's data member`vertexprops_` store the tagId.

Reply:
ACK.

Reply:
I think you can materialize `TagID` before you pass it to `Traverse` or `AppendVertices` instead of doing it in a visitor, which is debug friendly as well.

Reply:
I think both are ok, other visitors like `DeducePropsVisitor` has used `tagId`: 

```
void DeducePropsVisitor::visit(LabelTagPropertyExpression *expr) {
  auto status = qctx_->schemaMng()->toTagID(space_, expr->sym());
```